streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -609,6 +615,24 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestR
       } yield l.reverse must_=== (0 to 10).toList
     )
 
+  private def dropUntil = {
+    def dropUntil[A](as: List[A])(f: A => Boolean): List[A] =
+      as.dropWhile(!f(_)).drop(1)
+    prop { (s: Stream[String, Byte], p: Byte => Boolean) =>
+      unsafeRunSync(s.dropUntil(p).runCollect) must_=== unsafeRunSync(s.runCollect.map(dropUntil(_)(p)))
+    }
+  }
+
+  private def dropUntilShortCircuiting =
+    unsafeRun {
+      (Stream(1) ++ Stream.fail("Ouch"))
+        .take(1)
```

Comment:
you can take that one out, can't you?

Reply:
Actually, looking at it I am wondering if there is an issue with this and the existing `dropWhile` test. If we are only taking the first element we're never going to see the failure so we're not testing anything about the drop function, right?

Reply:
Hmm, yeah I agree 😄

Reply:
So thinking about this more I think the entire `dropWhile` short circuiting property is not sound. `DropWhile` has to look at each element to determine if it should drop it, and all the elements it doesn't look at are by definition not dropped. The only way to get some kind of "short circuiting" would be if `dropWhile` looked at each value to see if it was an error and then dropped those two while it was operating, which seems inconsistent with the other combinators. `takeWhile` makes sense because you are only taking a certain number of values, so of course you would ignore any errors that existed in values past the ones you were taking.

Reply:
The `filter` short circuiting properties have the same issue.